tests/all/test_values.rs

```diff
@@ -443,31 +443,34 @@ fn test_metadata() {
     // let vec_val = VectorType::const_vector(&[i8_val]);
     // let fn_val = module.add_function("my_fn", fn_type, None);
 
-    let md_node = context.metadata_node(&[bool_val.into(), f32_val.into()]);
+    let md_node_child = context.metadata_node(&[f32_val.into()]);
+    let md_node = context.metadata_node(&[bool_val.into(), f32_val.into(), md_string.into(), md_node_child.into()]);
 
     let node_values = md_node.get_node_values();
 
     assert_eq!(md_node.get_string_value(), None);
-    assert_eq!(node_values.len(), 2);
+    assert_eq!(node_values.len(), 4);
     assert_eq!(node_values[0].into_int_value(), bool_val);
     assert_eq!(node_values[1].into_float_value(), f32_val);
+    assert_eq!(node_values[2].into_metadata_value().get_string_value(), md_string.get_string_value());
+    assert!(node_values[3].into_metadata_value().is_node());
 
-    module.add_global_metadata("my_md", &md_string);
+    //module.add_global_metadata("my_md", &md_string);
```

Comment:
`add_global_metadata` should return a result rather than asserting, and this then can remain a test assertion of Err

Reply:
Fixed